Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM debian:stretch-slim
+
+ARG APACHE_SPARK_VERSION=2.2.0
+ARG HADOOP_VERSION=2.7
+ENV SPARK_NAME=spark-${APACHE_SPARK_VERSION}-bin-hadoop${HADOOP_VERSION}
+
+ENV SPARK_DIR /opt/${SPARK_NAME}
+ENV SPARK_HOME /usr/local/spark
+
+RUN mkdir -p /usr/share/man/man1 && \
+    apt-get update && \
+    apt-get install -y --no-install-recommends \
+    openjdk-8-jre-headless  ca-certificates-java curl && \
+    rm -rf /var/lib/apt/* && \
+    curl http://d3kbcqa49mib13.cloudfront.net/${SPARK_NAME}.tgz | \
+    tar xzf - -C /opt && \
+    mkdir -p /root/.sbt/launchers/0.13.13 && \
+    curl -L -o /root/.sbt/launchers/0.13.13/sbt-launch.jar http://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt-launch/0.13.13/sbt-launch.jar && \
```

Comment:
Could you help to understand, what is the be benefit of hard-coding it here (with version number), instead of letting ./sbt wrapper to take care of it below?

Reply:
by the time we run `./sbt` there is no curl anymore. It can't download jar and recommends to download it manually as I do here. I can just keep curl in container then we can remove it if you prefer that.

Reply:
If `sbt` needs `curl` to do its work, I'd consider `curl` as a `sbt` dependency and leave it in the container

Reply:
`sbt` itself doesn't need curl. the script to install `sbt` needs it. anyway i'll keep`curl` in container.